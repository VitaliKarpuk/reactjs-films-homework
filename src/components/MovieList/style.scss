.movie-list{
  max-width: 100%;
  background-color: rgb(228, 228, 228);
  padding: 50px 30px;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  align-items: flex-start;
}