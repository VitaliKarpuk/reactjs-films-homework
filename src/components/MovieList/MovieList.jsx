/* eslint-disable react/jsx-key */
import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import PropTypes from "prop-types";
import Film from "../Film/Film";
import requestFilmsByGenre from "../../modules/actions/requestFilmsByGenre";
import requesFilms from "../../modules/actions/requestFilms";
import { useParams } from "react-router";
import Loading from "../Loading/Loading";

import "./style.scss";

const MovieList = ({ films, genre, movieListView }) => {
  const { id: filmGenre, typeSearch } = useParams();
  const dispatch = useDispatch();
  const showListFilms = useSelector(state => state.showListFilms);
  const arrGenre = useSelector(state => state.genre);

  useEffect(() => {
    if (filmGenre && arrGenre) {
      dispatch(requestFilmsByGenre("popular", filmGenre, 1));
    } else {
      dispatch(requesFilms(typeSearch, 1));
    }
  }, [filmGenre, typeSearch, arrGenre]);

  const changeGenre = arr => {
    return arr.reduce((acc, item, index) => {
      genre.forEach(elem => {
        if (elem.id === item) {
          acc[index] = elem.name;
        }
      });
      return acc;
    }, []);
  };

  return (
    <section className="movie-list">
      {!showListFilms ? (
        <Loading />
      ) : (
        <div
          className={
            movieListView
              ? "movie-list__wrapper-items"
              : "movie-list__wrapper-items_active"
          }
        >
          {films.map(film => {
            return (
              <Film
                movieListView={movieListView}
                key={film.id}
                title={film.title}
                overview={film.overview}
                imgUrl={film.poster_path}
                rating={film.vote_average}
                genreFilmArray={film.genre_ids}
                genre={genre}
                changeGenre={changeGenre}
                id={film.id}
              />
            );
          })}
        </div>
      )}
    </section>
  );
};

MovieList.propTypes = {
  films: PropTypes.array,
  genre: PropTypes.array,
  movieListView: PropTypes.bool
};

MovieList.defaultProps = {
  films: [],
  genre: [],
  movieListView: true
};

export default MovieList;
