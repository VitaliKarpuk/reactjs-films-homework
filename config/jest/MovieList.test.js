/* eslint-disable no-undef */
import React from 'react'
import MovieList from '../../src/components/MovieList/MovieList';

import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import cofigureMockStore from 'redux-mock-store';


const mockStore = cofigureMockStore();

describe('My Connected React-Redux Component', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      films: [{ id: 1, genre_ids: [1] }],
      genre: [{ id: 1 }],
      dispatch: jest.fn()
    });


    component = renderer.create(
      <Provider store={store}>
        <MovieList />
      </Provider>
    );
  });

  it("renders Search component", () => {
    const result = component.toTree().rendered.rendered.props.children[0].props.changeGenre([{ id: 1, name: 'test' }])
    expect(result).toEqual([]);
  });

  it("renders Search component", () => {
    expect(component.toJSON()).toMatchSnapshot();;
  });
})