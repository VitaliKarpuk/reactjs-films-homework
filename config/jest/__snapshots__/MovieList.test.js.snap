// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component renders Search component 1`] = `null`;
